### ISSUE TITLE
Failed testcases in temporal analogy

### ISSUE THREAD

**Issue by l591326240** [opennars/opennars#34]
```
             IN: <<robin --> bird> </> <robin --> animal>>. 
             IN: <<robin --> [flying]> =/> <robin --> bird>>.

Expected output :

             OUT: <<robin --> [flying]> =/> <robin --> animal>>.
             OUT: <<robin --> animal> =\> <robin --> bird>>.

Actual:

            No output

---------------------------------------------------------------------------------------------

           IN: <<robin --> bird> </> <robin --> animal>>. 
           IN:<<robin --> bird> =\> <robin --> [flying]>>.

Expected output:

           OUT: <<robin --> [flying]> =/> <robin --> animal>>.
           OUT: <<robin --> animal> =\> <robin --> bird>>.

Actual:

           No output
           
```

**Comment by patham9** [comment]
this is a good find, I will have a look at this issue the next days.

Best regards,
Patrick

**Comment by automenta** [comment]
is this an existing testcase, or a variation of one?

btw the first one seems to work in volatile4. ill add the others as test cases if they arent already in the test set

![screenshot_2015-04-19_21-51-36](https://cloud.githubusercontent.com/assets/157093/7222513/3217108c-e6de-11e4-8fbb-2eeeb2974261.png)

**Comment by l591326240** [comment]
This is not an existing testcase. I made it base on an existing example of NAL5, so it should be a variation.

I just tried both of them again with 1.6.3, but still no outputs T_T.

btw automenta....are you Seth? :)

**Comment by automenta** [comment]
more test cases are always helpful, thanks!

**Comment by l591326240** [comment]
I've made a lot of testcases for NAL7, they are pretty much the variations of examples from existing testcases of NAL5. But I don't think those are enough, I should think about more examples so finally NARS could deal with all different kind of testcases.

I am testing NAL8....it doesn't look good so far...I think I will post some issues after meeting with Dr. Wang on Tuesday. 

btw, automenta, are you Seth ? :)

**Comment by automenta** [comment]
yes this is seth

in the development branch volatile4 there is a JUnit parametric testing framework.  so all the permutations of a range of test cases can be specified easily. i use this to also run the same test on different NAR parameters.

**Comment by l591326240** [comment]
Hi Seth, this is Xiang. 

Actually I have a question for you, for some nodes in the Concept Network, sometimes they correspond to some long statements so we can't see the entire statement. Just like the node in the following picture
![image](https://cloud.githubusercontent.com/assets/11791925/7222856/81c5d15c-e749-11e4-9919-31903725a543.png)

![image](https://cloud.githubusercontent.com/assets/11791925/7222839/38231730-e749-11e4-86fb-6e6b9376e372.png)

Is there a way we can see the entire statement, if not, could you fix it in 1.6.4, because I think it will be helpful for testing, thanks.

**Comment by automenta** [comment]
https://github.com/opennars/opennars/blob/1.6.4_devel/nars_gui/nars/gui/output/graph/NARGraphDisplay.java#L34 you can increase the length of the displayed text by increasing the label length variable. there may be other things you want to adjust there too

**Comment by l591326240** [comment]
Okay, got it, thank you!!!!

**Comment by patham9** [comment]
The first one works, it generates ```<<robin --> [flying]> =/> <robin --> animal>>. ```

But it won't generate ```<<robin --> animal> =\> <robin --> bird>>.``` for redundancy considerations.
Else it would have to transform between </> and =\> and =/> all the time, this only happens on the temporal induction. (For the same reason it won't generate result2 in the second example)

In the second example, the first valid expected output is
```
<<robin --> animal> =\> <robin --> [flying]>>.
```
which 1.6.4-devel generates, and not
```
 <<robin --> [flying]> =/> <robin --> animal>>.
```

So this issue seems to be solved in 1.6.4-devel already without any additional demanded changes.

Best regards,
Patrick
